Cleanups, fixes, Makefile, …

liskin opened the pull request:
#### [Clean up Gemfile and _config.yml](../commit/62bbc7d44cb43b71e2af24a8dc5340227626f5c4)

Drop plugins that are included in github-pages by default, drop stuff
that isn't needed/used (minima, Windows-specific gems), drop baseurl.

The thinning of Gemfile may actually be necessary, as bundler's
dependency resolution sometimes installs a horribly old version of
github-pages when it starts the resolution with jekyll-default-layout,
as github-pages depends on an older version of jekyll-default-layout,
and its resolution algorithm then goes all crazy.

#### [Add Makefile to install github-pages and build/serve the site](../commit/febf3f021ed0b38c47bc41bc786625934ac55311)


#### [Exclude dev plumbing from _site](../commit/399e957b8b1aab0507a1f5b9909dedb9ade3663d)


#### [Drop unused css/default.css](../commit/4d8b02db9bf6db04517fb020d4633cd8fc392aca)


#### [Speed up Jekyll build](../commit/38fb561f2ac5da700745054a6e9af06235bea25f)

We're not using any Jekyll themes and this cuts the build time from 1.5
seconds to a little over 100 milliseconds.

#### [Fix Raleway font not being used](../commit/3e5253b8a728af810c28ebd01638e8b3b2684d0d)


#### [Fix broken links to non-existent intro.html](../commit/056c9bba11ed2040767091ac719b2c468f974b9f)

What used to be intro.html is now install-instructions.md.

#### [install-instructions: Refresh the list of required libraries](../commit/bcdf70daacc9c523ac78e5602cf7005766084ae3)


#### [tour: Fix unclosed `](../commit/44d1a2588598e1b376280ff5fdde5c22efc3d434)


#### [Workaround for livereload @import bug](../commit/d50b38e4c9f8bd580aa9e87199d0a98722a68ed0)

Gets rid of the annoying

    [2021-06-04 10:27:26] ERROR `/font-raleway.css' not found.

because livereload incorrectly handles relative @import in CSS.

Related: https://github.com/lepture/python-livereload/issues/248

#### [Relative links to images](../commit/0888cc30a8fa6accfd69d25ad9322b18d8efb0ea)

site.url is hardcoded to https://xmonad.org, even when serving locally
or from a fork at GitHub.

#### [Add GitHub Actions workflow to check links (both internal and external)](../commit/364c571661da937ba0133d12833bab40bebe2fc6)

"Stolen" from https://github.com/liskin/work.lisk.in/blob/c6ad22896ad933aed93cc752e0373d4bdb3bc3a8/.github/workflows/linkchecker.yaml

#### [Fix/add links to former xmonad developers](../commit/7c91d5204a7470f068a354a209ee6f5672beec55)


#### [Fix broken links to dzen and rxvt-unicode](../commit/f0d13ffe2b37b8cb8eb6183a855f0d62ef3efa8d)

liskin committed:
Clean up Gemfile and _config.yml

Drop plugins that are included in github-pages by default, drop stuff
that isn't needed/used (minima, Windows-specific gems), drop baseurl.

The thinning of Gemfile may actually be necessary, as bundler's
dependency resolution sometimes installs a horribly old version of
github-pages when it starts the resolution with jekyll-default-layout,
as github-pages depends on an older version of jekyll-default-layout,
and its resolution algorithm then goes all crazy.

liskin committed:
Add Makefile to install github-pages and build/serve the site

liskin committed:
Exclude dev plumbing from _site

liskin committed:
Drop unused css/default.css

liskin committed:
Speed up Jekyll build

We're not using any Jekyll themes and this cuts the build time from 1.5
seconds to a little over 100 milliseconds.

liskin committed:
Fix Raleway font not being used

liskin committed:
Fix broken links to non-existent intro.html

What used to be intro.html is now install-instructions.md.

liskin committed:
install-instructions: Refresh the list of required libraries

liskin committed:
tour: Fix unclosed `

liskin committed:
Workaround for livereload @import bug

Gets rid of the annoying

    [2021-06-04 10:27:26] ERROR `/font-raleway.css' not found.

because livereload incorrectly handles relative @import in CSS.

Related: https://github.com/lepture/python-livereload/issues/248

liskin committed:
Relative links to images

site.url is hardcoded to https://xmonad.org, even when serving locally
or from a fork at GitHub.

liskin committed:
Add GitHub Actions workflow to check links (both internal and external)

"Stolen" from https://github.com/liskin/work.lisk.in/blob/c6ad22896ad933aed93cc752e0373d4bdb3bc3a8/.github/workflows/linkchecker.yaml

liskin committed:
Fix/add links to former xmonad developers

liskin committed:
Fix broken links to dzen and rxvt-unicode

liskin commented:
GitHub says the branch hasn't been deployed but it has, here: https://liskin.github.io/xmonad-web/tour.html

slotThe commented on download.md:
```diff
@@ -71,6 +71,6 @@ cabal install https://github.com/xmonad/xmonad-contrib/archive/master.tar.gz
 *   [dmenu](https://tools.suckless.org/dmenu/), a program launcher
-*   [dzen](https://gotmor.googlepages.com/dzen), an extensible status bar
+*   [dzen](https://robm.github.io/dzen/), an extensible status bar
 *   [xmobar](https://hackage.haskell.org/package/xmobar), an extensible status bar
```
I wonder of hackage is the right place to link to concerning xmobar.  This should probably go to either GitHub or `xmobar.org`

liskin replied:
xmobar.org seems to be from before https://github.com/jaor/xmobar/commit/fa8c748b5b2e3a1a6f6add0ad6bc2a8e56ddcc28, which seems to also be the point when Hackage stopped rendering xmobar's readme, so yeah, linking to GitHub is now the best option.

slotThe reviewed (approved):
I like this!

Can confirm steps in the README work for me when building with the new Makefile

TheMC47 reviewed (approved):
Thank you for this much-needed cleanup! At least some of us have a clue on what they're doing :grinning:

liskin committed:
Update link to xmobar

Since https://github.com/jaor/xmobar/commit/fa8c748b5b2e3a1a6f6add0ad6bc2a8e56ddcc28,
Hackage no longer renders the readme, and https://xmobar.org/ also
doesn't seem to be updated since then.
